ui/src/components/app-form/app-form.tsx

```diff
@@ -228,6 +429,13 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
         },
         keep_alive: keepAlive,
       };
+
+      // Add repository details only if it's a Git-based app
+      if (deployOption === 'git') {
+        payload.repository = { url: gitUrl };
+      }
+      // After successful submission, start checking the app status
+      checkAppStatus(displayName, currentUser?.name || '');
```

Comment:
This check is not valid here. In this branch, since there are profiles, the user would be directed to the server types page, which means the app hasn't been submitted for creation yet.

I'm guessing the `checkAppStatus` function can be removed entirely.

Reply:
Verified, and removed checkAppStatus